File: ViewPagerAndFragment/app/src/main/java/rahulpahuja/com/viewpagerandfragment/activity/MainActivity.java
package rahulpahuja.com.viewpagerandfragment.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import rahulpahuja.com.viewpagerandfragment.R;

public class MainActivity extends AppCompatActivity {
    /*Creating an intent to start the activity with desired selection*/
    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        /*Fetching the Buttons*/
        Button pagerSimple = (Button) findViewById(R.id.btn_simple_pager);
        Button pagerTab = (Button) findViewById(R.id.btn_stripped_pager_tab);
        Button pagerTitle = (Button) findViewById(R.id.btn_stripped_pager_title);

        /*Setting the Click Listener*/
        pagerSimple.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent = new Intent(MainActivity.this, ViewPagerActivity.class);
                startActivity(intent);
            }
        });

        /*Setting the Click Listener*/
        pagerTab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent = new Intent(MainActivity.this, ViewPagerTabStripActivity.class);
                startActivity(intent);
            }
        });

        /*Setting the Click Listener*/
        pagerTitle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                intent = new Intent(MainActivity.this, ViewPagerTitleStripActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: LifecycleAwareComponents/app/src/main/java/com/rahulandroid/lifecycleawarecomponents/MainActivity.kt
package com.rahulandroid.lifecycleawarecomponents

import android.content.Intent
import android.content.res.Configuration
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

class MainActivity : AppCompatActivity() {

    private final var TAG = "My App MainAcitivy"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        lifecycle.addObserver(MainActivityObserver("My App "+"MainActivityObserver"))
        Log.d(TAG, "onCreate() called with: savedInstanceState = $savedInstanceState")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart() called")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume() called")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause() called")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop() called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy() called")
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        Log.d(TAG, "onConfigurationChanged() called with: newConfig = $newConfig")
    }

    override fun onBackPressed() {
        super.onBackPressed()
        Log.d(TAG, "onBackPressed() called")
    }

    fun onClick(view: View) {
        Log.d(TAG, "onClick() called with: view = $view")
        startActivity(Intent(this,MainActivity2::class.java))
    }
}